server/src/main/java/com/google/android/attestation/ParsedAttestationRecord.java

```diff
@@ -96,9 +96,27 @@ private ParsedAttestationRecord(
     this.teeEnforced = teeEnforced;
   }
 
-  public static ParsedAttestationRecord createParsedAttestationRecord(X509Certificate cert)
+  public static ParsedAttestationRecord createParsedAttestationRecord(X509Certificate[] certs)
       throws IOException {
-    ASN1Sequence extensionData = extractAttestationSequence(cert);
+
+    // The last seen certificate with an attestation is considered benign (i.e.,
+    // generated and signed in the TEE). It is the one closest to the root
+    // certificate and thus could not have been forged without an attacker
+    // knowing the private key of the certificates above it.
+    byte[] attestationExtensionBytes = null;
+    int i;
+    for (i = certs.length - 1; i >= 0; i--) {
+      attestationExtensionBytes = certs[i].getExtensionValue(KEY_DESCRIPTION_OID);
+      if (attestationExtensionBytes != null && attestationExtensionBytes.length != 0) {
```

Comment:
Alternative code:
for (int i = certs.length - 1; i >= 0; i--) {
  byte[] attestationExtensionBytes = certs[i].getExtensionValue(KEY_DESCRIPTION_OID);
  if (attestationExtensionBytes != null && attestationExtensionBytes.length != 0) {
    return new ParsedAttestationRecord(extractAttestationSequence(attestationExtensionBytes));
  }
}
throw new IllegalArgumentException("Couldn't find the keystore attestation extension data.");